File: app/src/test/java/com/domclick/repository/AccountRepositoryTest.java
package com.domclick.repository;

import com.domclick.BaseTestSupport;
import com.domclick.app.AppApplication;
import com.domclick.model.Account;
import com.domclick.model.User;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.math.BigDecimal;
import java.util.Optional;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = AppApplication.class)
public class AccountRepositoryTest extends BaseTestSupport {
    private User user;

    @Before
    public void init() {
        user = userRepository.findById(0L).orElseThrow(
                () -> new IllegalStateException("Datasource do not initialize"));
    }

    @Test
    public void crudTest() {
        Account account = createTestAccount(user);

        account = accountRepository.findById(account.getId()).orElseThrow(() -> new RuntimeException("It's impossible"));
        account.setBalance(new BigDecimal(50000d));
        account = accountRepository.save(account);
        account = accountRepository.findById(account.getId()).orElseThrow(() -> new RuntimeException("It's impossible"));
        Assert.assertEquals(account.getBalance(), newBigDecimal(50000d));

        accountRepository.delete(account);
        Optional<Account> accountOptional = accountRepository.findById(account.getId());
        Assert.assertTrue(!accountOptional.isPresent());
    }
}

File: app/src/test/kotlin/com/domclick/repository/AccountRepositoryTest.kt
package com.domclick.repository

import com.domclick.BaseTestSupport
import com.domclick.app.AppApplication
import org.junit.runner.RunWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner

@RunWith(SpringRunner::class)
@SpringBootTest(classes = [(AppApplication::class)])
class AccountRepositoryTest : BaseTestSupport() {
    /*private lateinit var user: User

    //Todo refactoring

    @Before
    fun init() {
        user = userRepository.findById(0L).orElseThrow { IllegalStateException("Datasource do not initialize") }
    }

    @Test
    fun crudTest() {
        var account = createTestAccount(user)

        account = accountRepository.findById(account.id).orElseThrow { RuntimeException("It's impossible") }
        account.balance = BigDecimal(50000.0)
        account = accountRepository.save(account)
        account = accountRepository.findById(account.id).orElseThrow { RuntimeException("It's impossible") }
        Assert.assertEquals(account.balance, newBigDecimal(50000.0))

        accountRepository.delete(account)
        val accountOptional = accountRepository.findById(account.id)
        Assert.assertTrue(!accountOptional.isPresent())
    }*/
}
